File: exercicios/2221/2221.py
n = int(input())
vg = 0
a,d,l = 0,0,0
vg = []
for i in range(n):
    b = int(input())

    for j in range(2):

        x = input().split()
        a = int(x[0])
        d = int(x[1])
        l = int(x[2])
        vg.append(j)

        vg[j] = ((a + d)/ 2 )

        if l % 2 == 0:
            vg[j] += b
        
    if vg[0] < vg[1]:
        print('Guarte')
    elif vg[0] == vg[1]:
        print('Empate')
    elif vg[0] > vg[1]:
        print('Dabriel')

        

File: exercicios/JavaScript/2221/2221.js
var input = require('fs').readFileSync('stdin','utf8');
var lines = input.split('\n');

var t = parseInt(lines.shift());

for(let i=0; i<t;i++){
    var b = parseInt(lines.shift())

    var a1,d1,l1,a2,d2,l2 = 0;

    var golpe1,golpe2 = 0;

    var [a1,d1,l1] = lines.shift().split(" ").map(Number);

    var [a2,d2,l2] = lines.shift().split(" ").map(Number);


    golpe1 = ((parseInt(a1) + parseInt(d1))/2)
    golpe2 = ((parseInt(a2) + parseInt(d2))/2)

    if(l1 % 2 == 0){
        golpe1 += b;
    }if(l2 % 2 == 0){
        golpe2 += b
    }

    if(golpe1 > golpe2){
        console.log('Dabriel');
    }else if(golpe1 < golpe2){
        console.log('Guarte');
    }else{
        console.log('Empate')
    }
}